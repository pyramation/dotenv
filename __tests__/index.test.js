import dotenv from '../src';
import { resolve } from 'path';
import cases from 'jest-in-case';

cases(
  'dotenv',
  opts => {
    const path = resolve(__dirname + '/../__fixtures__/' + opts.test);
    process.env.NODE_ENV = opts.name;
    dotenv(path);
    const env = Object.keys(process.env).reduce((m, key) => {
      if (key.match(/^DOTENV_TEST_/)) {
        m[key] = process.env[key];
      }
      return m;
    }, {});
    expect(env).toMatchSnapshot();
    // after each clear env...
    Object.keys(process.env).forEach(key => {
      if (key.match(/^DOTENV_TEST_/)) {
        delete process.env[key];
      }
    });
  },
  [
    { name: 'development', test: 'test1' },
    { name: 'test', test: 'test1' },
    { name: 'production', test: 'test1' }
  ]
);
