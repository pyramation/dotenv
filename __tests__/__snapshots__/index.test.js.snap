// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotenv development 1`] = `
Object {
  "DOTENV_TEST_BASE_ENV": "1",
  "DOTENV_TEST_BASE_ENV_LOCAL": "1",
  "DOTENV_TEST_ENV_DEV": "1",
  "DOTENV_TEST_ENV_DEV_LOCAL": "1",
}
`;

exports[`dotenv production 1`] = `
Object {
  "DOTENV_TEST_BASE_ENV": "1",
  "DOTENV_TEST_BASE_ENV_LOCAL": "1",
  "DOTENV_TEST_ENV_PROD": "1",
  "DOTENV_TEST_ENV_PROD_LOCAL": "1",
}
`;

exports[`dotenv test 1`] = `
Object {
  "DOTENV_TEST_BASE_ENV": "1",
  "DOTENV_TEST_BASE_ENV_LOCAL": "1",
  "DOTENV_TEST_ENV_TEST": "1",
  "DOTENV_TEST_ENV_TEST_LOCAL": "1",
}
`;
